### PR TITLE
docs: update exceptions for sprintf and vsprintf

### PR DESCRIPTION
**Pull request by kesselb**
### vsprintf

image

### sprintf

image

### CODE REVIEW

File: reference/strings/functions/sprintf.xml [hunk 1]
```diff
@@ -43,6 +43,22 @@
...
+  <refsect1 role="errors">
+  &reftitle.errors;
+  <para>
+   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if the number of arguments is zero. Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
+  </para>
+  <para>
+   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if [width] is less than zero or bigger than <constant>PHP_INT_MAX</constant>. Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
+  </para>
+  <para>
+   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if [precision] is less than zero or bigger than <constant>PHP_INT_MAX</constant>. Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
```
Comment (Girgias) [inline reference/strings/functions/sprintf.xml:55]: Maybe?
```suggestion
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if <literal>[width]</literal> is less than zero or bigger than <constant>PHP_INT_MAX</constant>. Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
  </para>
  <para>
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if <literal>[precision]</literal> is less than zero or bigger than <constant>PHP_INT_MAX</constant>. Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
```

Comment (kesselb) [reply reference/strings/functions/sprintf.xml]: Good suggestion, thank you! I just updated the pull request.

**Review by Girgias — COMMENTED** [review]
Actually doesn't this apply to all functions which take such a format? In which case creating a new XML entity in language-snippets might make more sense

**Comment by kesselb** [comment]
> Actually doesn't this apply to all functions which take such a format? In which case creating a new XML entity in language-snippets might make more sense

Good idea.  The only difference is ValueError and ArgumentCountError. 

ArgumentCountError: sprintf, printf, fprintf
ValueError: vprintf, vfprintf, vsprintf, 

I will take a look.

**Comment by kesselb** [comment]
Moved errors and changelog to language-snippets.

File: language-snippets.ent [hunk 2]
```diff
@@ -3833,6 +3826,134 @@ local: {
...
+      <entry>8.0.0</entry>
+      <entry>
+       Throw a <classname>ArgumentCountError</classname> when less arguments are given than required;
+       previously this function emitted a <constant>E_WARNING</constant> instead.
+      </entry>
+     </row>
+    </tbody>
+   </tgroup>
+  </informaltable>
+'>
```
Comment (Girgias) [inline language-snippets.ent:3876]: Sorry my mistake, we generally don't make the whole changelog table tag an entity as this makes it hard to update something *specific* on a function, but maybe for sprintf and co this is not too much a problem.

@mumumu and @saundefined what do you think?

Comment (saundefined) [reply language-snippets.ent:3876]: @Girgias LGTM

File: language-snippets.ent [hunk 3]
```diff
@@ -3833,6 +3826,134 @@ local: {
...
+  <para xmlns="http://docbook.org/ns/docbook">
+   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if <literal>[width]</literal> is less than zero or bigger than <constant>PHP_INT_MAX</constant>. Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
+  </para>
+  <para xmlns="http://docbook.org/ns/docbook">
+   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if <literal>[precision]</literal> is less than zero or bigger than <constant>PHP_INT_MAX</constant>. Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
+  </para>
+  <para xmlns="http://docbook.org/ns/docbook">
+   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown when less arguments are given than required. Prior to PHP 8.0.0, &false; was returned and a <constant>E_WARNING</constant> emitted instead.
+  </para>
+'>
```
Comment (Girgias) [inline language-snippets.ent:3955]:
```suggestion
<!ENTITY strings.errors.sprintf '
  <para xmlns="http://docbook.org/ns/docbook">
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if the number of arguments is zero.
   Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
  </para>
  <para xmlns="http://docbook.org/ns/docbook">
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if <literal>[width]</literal> is less than zero or bigger than <constant>PHP_INT_MAX</constant>.
   Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
  </para>
  <para xmlns="http://docbook.org/ns/docbook">
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if <literal>[precision]</literal> is less than zero or bigger than <constant>PHP_INT_MAX</constant>.
   Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
  </para>
  <para xmlns="http://docbook.org/ns/docbook">
   As of PHP 8.0.0, a <classname>ArgumentCountError</classname> is thrown when less arguments are given than required.
   Prior to PHP 8.0.0, &false; was returned and a <constant>E_WARNING</constant> emitted instead.
  </para>
'>

<!ENTITY strings.errors.vsprintf '
  <para xmlns="http://docbook.org/ns/docbook">
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if the number of arguments is zero.
   Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
  </para>
  <para xmlns="http://docbook.org/ns/docbook">
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if <literal>[width]</literal> is less than zero or bigger than <constant>PHP_INT_MAX</constant>.
   Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
  </para>
  <para xmlns="http://docbook.org/ns/docbook">
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown if <literal>[precision]</literal> is less than zero or bigger than <constant>PHP_INT_MAX</constant>.
   Prior to PHP 8.0.0, a <constant>E_WARNING</constant> was emitted instead.
  </para>
  <para xmlns="http://docbook.org/ns/docbook">
   As of PHP 8.0.0, a <classname>ValueError</classname> is thrown when less arguments are given than required.
   Prior to PHP 8.0.0, &false; was returned and a <constant>E_WARNING</constant> emitted instead.
  </para>
'>
```

File: reference/strings/functions/fprintf.xml [hunk 4]
```diff
@@ -48,6 +48,16 @@
    Returns the length of the string written.
   </para>
  </refsect1>
+
```
Comment (Girgias) [inline reference/strings/functions/fprintf.xml:51]:
```suggestion

```

**Comment by Girgias** [comment]
Thank you!

**Pull request merged**